hudi-common/src/main/java/org/apache/hudi/common/table/TableSchemaResolver.java

```diff
@@ -109,13 +110,18 @@ private MessageType getTableParquetSchemaFromDataFile() {
           // Determine the file format based on the file name, and then extract schema from it.
           if (instantAndCommitMetadata.isPresent()) {
             HoodieCommitMetadata commitMetadata = instantAndCommitMetadata.get().getRight();
-            String filePath = commitMetadata.getFileIdAndFullPaths(metaClient.getBasePath()).values().stream().findAny().get();
-            if (filePath.contains(HoodieFileFormat.HOODIE_LOG.getFileExtension())) {
-              // this is a log file
-              return readSchemaFromLogFile(new Path(filePath));
-            } else {
-              return readSchemaFromBaseFile(filePath);
+            Iterator<String> filePaths = commitMetadata.getFileIdAndFullPaths(metaClient.getBasePath()).values().iterator();
+            MessageType type = null;
+            while (filePaths.hasNext() && type == null) {
```

Comment:
Let's unify this behavior across both MOR/COW

Reply:
No problem, but this NPE may only happen in `.log` file

Reply:
Understood. But behavior should be consistent regardless of whether this is COW or MOR

Reply:
Ok. Thanks for reviewing, and the new commit has fixed it.

Reply:
guess Alexey's suggestion was to introduce a private method and call from both places.

Reply:
> guess Alexey's suggestion was to introduce a private method and call from both places.

Got it, thanks for your suggestion.